pkg/route/ingress/ingress.go

```diff
@@ -623,7 +623,7 @@ func newRouteForIngress(
 				Name: path.Backend.Service.Name,
 			},
 			Port:           port,
-			TLS:            tlsConfigForIngress(ingress, rule, tlsSecret),
+			TLS:            tlsConfigForIngress(ingress, rule, tlsSecret, secretLister),
```

Comment:
It seems a little strange to add `secretLister` to `tlsConfigForIngress`'s parameters, with the result that the caller uses `secretLister` to get `tlsSecret` and then passes both `tlsSecret` and `secretLister` to `tlsConfigForIngress`.  Would one of the following refactorings make sense?

* Have the caller get `destinationCACertificate` and pass it instead of `secretLister` to `tlsConfigForIngress`, xor
* have the caller pass `secretLister` to `tlsConfigForIngress`, and then have `tlsConfigForIngress` look up both `tlsSecret` and `destinationCACertificate`.

The latter would require also changing `tlsConfigForIngress` to return `hasInvalidTLSSecret` and would require some minor shuffling, but I imagine it would result in slightly less code duplication in the end.

Reply:
I Agree. I applied the second way.